pubsub/azure/eventhubs/eventhubs.go

```diff
@@ -162,15 +423,62 @@ func (aeh *AzureEventHubs) Init(metadata pubsub.Metadata) error {
 	if err != nil {
 		return err
 	}
-	userAgent := "dapr-" + logger.DaprVersion
+
 	aeh.metadata = m
-	hub, err := eventhub.NewHubFromConnectionString(aeh.metadata.connectionString,
-		eventhub.HubWithUserAgent(userAgent))
-	if err != nil {
-		return fmt.Errorf("unable to connect to azure event hubs: %v", err)
+	aeh.eventProcessors = map[string]*eph.EventProcessorHost{}
+	aeh.hubClients = map[string]*eventhub.Hub{}
+
+	if aeh.metadata.ConnectionString != "" {
+		// Validate connectionString.
+		hubName, err := validateAndGetHubName(aeh.metadata.ConnectionString)
+		if err != nil {
+			return errors.New(invalidConnectionStringErrorMsg)
+		}
+		if hubName != "" {
+			aeh.logger.Infof("connectionString will be used to connect to given hub %q only. connection established on first publish/subscribe", hubName)
+			aeh.logger.Debugf("req.Topic field in incoming requests must be same as the one in the connectionString")
+		} else {
+			aeh.logger.Infof("hubName not given in connectionString. connection established on first publish/subscribe")
+			aeh.logger.Debugf("req.Topic field in incoming requests honored")
+		}
+		if aeh.metadata.EnableEnitiyManagement {
+			// See https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-management-libraries
+			return errors.New(entityManagementConnectionStrMsg)
+		}
+	} else {
+		// Connect via AAD.
+		settings, sErr := azauth.NewEnvironmentSettings("eventhubs", metadata.Properties)
+		if sErr != nil {
+			return sErr
+		}
+		aeh.eventHubSettings = settings
+		tokenProvider, err := aeh.eventHubSettings.GetAADTokenProvider()
+		if err != nil {
+			return fmt.Errorf("%s %s", hubManagerCreationErrorMsg, err)
+		}
+		aeh.tokenProvider = tokenProvider
+		aeh.logger.Info("connecting to Azure EventHubs via AAD. connection established on first publish/subscribe")
+		aeh.logger.Debugf("req.Topic field in incoming requests honored")
+
+		if aeh.metadata.EnableEnitiyManagement {
```

Comment:
A couple of more points here: 
if entity Management is enabled, it does need permissions at least at the EventHub namespace level to be able to create event hubs and consumer groups. 
Also the other metadata, SubscriptionID and ResourceGroupName is a mandatory requirement when enabling entityManagement as for now, it assumes that the topic will be subscribed to. 

If needed resourceGroupName and SubscriptionID can be validated only as part of subscription metadata validation since it will only be required for a subscriber and not for a publisher alone.

Reply:
wrt comment https://github.com/dapr/components-contrib/pull/1055#issuecomment-892832206 from @CodeMonkeyLeet , entity management for EventHubs through Dapr does step into provisioning infrastructure resources.